src/libsync/networkjobs.cpp

```diff
@@ -328,21 +337,21 @@ bool LsColJob::finished()
         connect( &parser, SIGNAL(directoryListingIterated(const QString&, const QMap<QString,QString>&)),
                  this, SIGNAL(directoryListingIterated(const QString&, const QMap<QString,QString>&)) );
         connect( &parser, SIGNAL(finishedWithError(QNetworkReply *)),
-                 this, SIGNAL(finishedWithError(QNetworkReply *)) );
+                 this, SIGNAL(parserFinishedWithError(QNetworkReply *)) );
```

Comment:
It isn't right now, because the parser does its job in one pass synchronously, and the LsColJob knows about errors through LsColXMLParser::parse() return code.
```
 if( !parser.parse( reply()->readAll(), &_sizes, expectedPath ) ) {
            // XML parse error
            emit finishedWithError()
```
But from the comment below if/when the parser works asynchronously, the signal will be needed. So it's better to keep it imho.
```
// TODO: Instead of doing all in this slot, we should iteratively parse in readyRead(). This
// would allow us to be more asynchronous in processing while data is coming from the network,
// not all in one big blob at the end.
```

I'll correct the signal/slot stuff.